components/bigqueryDS.js

```diff
@@ -0,0 +1,250 @@
+    this.tablePrefix = this.dataset + "." + this.datasetPrefix;
+    this.datasetLocation = argv.datasetLocation;
+    this.startTime = new Date(argv.startTime).toISOString();
+    this.endTime = new Date(argv.endTime).toISOString();
+    this.timeClause = "timestamp < @endTime and timeStamp > @startTime";
+  }
+
+  async bq(query, params) {
+    const options = { query, params, location: this.datasetLocation };
+    try {
+      const [rows] = await this.bigquery.query(options);
+      return rows;
+    } catch (err) {
+      if (err && err.message.match("Not found: Table")) {
+        console.error(
+          "Assuming missing table is due to odrd or lmfs not being enabled:",
+          err.message
+        );
+        return [];
+      }
+      throw err;
+    }
+  }
+  /*
+   * Extract trip_ids from vehicle logs, and query again to
+   * get the createTrip/update trip calls that may not be labeled
+   * with a vehicle
+   */
+  async fetchTripLogsForVehicle(vehicle_id, vehicleLogs) {
+    console.log("Loading trip logs for vehicle id", vehicle_id);
+    const trip_ids = _(vehicleLogs)
+      .map((x) => _.split(_.get(x, "labels.trip_id"), ","))
+      .flatten()
+      .uniq()
+      .compact()
+      .value();
+    const tripPromises = trip_ids.map(async (trip_id) => {
+      // TODO use a subquery and let BQ do all the work
+      const createLogs = await this.bq(
+        `select * from \`${this.tablePrefix}create_trip\` where labels.trip_id=@trip_id and ${this.timeClause} order by timestamp desc limit 10`,
+        {
+          trip_id: trip_id,
+          startTime: this.startTime,
+          endTime: this.endTime,
+        }
+      );
+      const updateLogs = await this.bq(
+        `select * from \`${this.tablePrefix}update_trip\` where labels.trip_id=@trip_id and ${this.timeClause} order by timestamp desc limit 10`,
+        {
+          trip_id: trip_id,
+          startTime: this.startTime,
+          endTime: this.endTime,
+        }
+      );
+      return [createLogs, updateLogs];
+    });
+
+    const tripLogs = _(await Promise.all(tripPromises))
+      .flatten()
+      .flatten()
+      .map(normalize)
+      .value();
+
+    console.log(`found ${tripLogs.length} ODRD trip logs`);
+    return tripLogs;
+  }
+
+  /*
+   * query update task to find tasks assigned to this vehicle and
+   * then use that result as subquery to find full list of create and
+   * update task requests associated with the specified vehicle.
+   *
+   * Note labels.delivery_vehicle_id is not necessarily set at creation
+   * time (or even on all update requests).
+   */
+  async fetchTaskLogsForVehicle(delivery_vehicle_id) {
+    console.log(
+      "Loading tasks logs for deliveryVehicle id",
+      delivery_vehicle_id
+    );
+    const params = {
+      delivery_vehicle_id,
+      //startTime: this.startTime,
+      //endTime: this.endTime,
+      // XXX ignore start time for field day
```

Comment:
Undo?  The values commented out are the command line defaults already no?

Or dump a WARNING log notifying user that the start/end time flags on command line are ignored?

Reply:
removed.